src/ai/backend/manager/models/image.py

```diff
@@ -269,12 +270,31 @@ def trimmed_digest(self) -> str:
 
     @property
     def image_ref(self) -> ImageRef:
+        if self.is_local:
+            image_name, tag = ImageRef.parse_image_tag(self.name)
+
+            return ImageRef(
+                image_name,
+                self.project,
+                tag,
+                self.registry,
+                self.architecture,
+                self.is_local,
+            )
+
         # Empty image name
         if self.project == self.image:
             image_name = ""
             _, tag = ImageRef.parse_image_tag(self.name.split(f"{self.registry}/", maxsplit=1)[1])
         else:
-            image_and_tag = self.name.split(f"{self.registry}/{self.project}/", maxsplit=1)[1]
+            join = functools.partial(join_non_empty, sep="/")
+            if self.is_local:
+                image_and_tag = self.name
+            else:
+                image_and_tag = self.name.split(
+                    f"{join(self.registry, self.project)}/", maxsplit=1
+                )[1]
```

Comment:
It seems like a good idea to define `join` only where you use it.
Additionally, I think `removeprefix` would be clearer.